lib/react_client/react_interop.dart

```diff
@@ -194,8 +344,13 @@ class ReactPortal {
 @JS()
 @anonymous
 class ReactComponent {
+  // TODO: Cast as Component2 in 6.0.0
   external Component get dartComponent;
+  // TODO how to make this JsMap without breaking stuff?
```

Comment:
Has this ship sailed?

Reply:
Offline: we have worked around this by making `InteropProps` implement `JsMap` for now.  Might be worth re-visiting in 6.0.0 though.